### PR TITLE
Implement contact form go api

### PR DESCRIPTION
**Pull request by cp-shruti-s**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1,16 +1,73 @@
...
+	}		
+
+	c.JSON(http.StatusOK, input)
+}
+
+
 func main() {
     router := gin.Default()
 
+	router.Use(static.Serve("/", static.LocalFile("./vue-frontend/dist", false)))
```
Comment (jimmy0251) [inline main.go:67]: We are not going to serve vue front-end with go anymore, so maybe we can remove this.

File: main.go [hunk 2]
```diff
@@ -1,16 +1,209 @@
...
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ses"
+	"github.com/gin-contrib/static"
 	"github.com/gin-gonic/gin"
+	"github.com/gin-gonic/gin/binding"
+	"github.com/markbates/pkger"
 )
 
+type ContactDetails struct {
```
Comment (cp-sumi-k) [inline main.go:26]: Run `go fmt .` for formatting go code

File: main.go [hunk 3]
```diff
@@ -1,16 +1,209 @@
...
+	if err != nil {
+		panic(err)
+	}
+
+	emailTemplate := template.Must(template.New("email-template.html").Parse(string(emailFile)))
+
+	return &ContactRepository{ HtmlTemplate: emailTemplate}
+}
+
+func (repository *ContactRepository) contactDetail(c *gin.Context) {
```
Comment (cp-sumi-k) [inline main.go:62]: I think we have to separate code in the `contact` directory, Instead of all in the main.go file

File: main.go [hunk 4]
```diff
@@ -1,16 +1,209 @@
...
+type ContactRepository struct {
+	HtmlTemplate *template.Template
+}
+
+const CHARSET = "utf-8"
+const SENDER = "contact@canopas.com"
+const RECEIVER = "hr@canopas.com"
+
+func New() *ContactRepository {
+	emailHtml, err := pkger.Open("/templates/email-template.html")
```
Comment (cp-sumi-k) [inline main.go:47]: We have to use `embed` instead of `pkger`

File: main.go [hunk 5]
```diff
@@ -1,16 +1,209 @@
...
 	"github.com/gin-gonic/gin"
+	"github.com/gin-gonic/gin/binding"
+	"github.com/markbates/pkger"
 )
 
+type ContactDetails struct {
+	Name 				string 				`json:"name"`
+	DesignationValue 	string 				`json:"designation_value"`
+	Business 			string 				`json:"business"`
+	CurrentWebsiteUrl 	map[string]string 	`json:"current_website_url"`
```
Comment (cp-sumi-k) [inline main.go:30]: Let's rename it to `SocialMediaLinks`

File: main.go [hunk 6]
```diff
@@ -1,16 +1,209 @@
...
+
+//go:embed vue-frontend/dist
+var frontendStatic embed.FS
+
 func main() {
     router := gin.Default()
 
+	router.Use(static.Serve("/", EmbedFolder(frontendStatic, "vue-frontend/dist")))
+
+	router.POST("/api/contact-detail", New().contactDetail)
```
Comment (cp-sumi-k) [inline main.go:205]: Let's rename endpoint to `send-contact-mail`

File: contact/contact.go [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+	c.JSON(http.StatusOK, input)
+}
+
+func (repository *Template) getEmailTemplate(input ContactDetails) (template *ses.SendEmailInput) {
+	SENDER := os.Getenv("SENDER")
+	RECEIVER := os.Getenv("RECEIVER")
+
+	htmlBody := repository.getHTMLBodyOfEmailTemplate(input)
+
+	fmt.Println(htmlBody)
```
Comment (cp-sumi-k) [inline contact/contact.go:77]: Let's remove log

File: go.mod [hunk 8]
```diff
@@ -10,12 +10,14 @@ require (
 require (
 	github.com/aws/aws-lambda-go v1.28.0 // indirect
 	github.com/gin-contrib/sse v0.1.0 // indirect
+	github.com/gin-contrib/static v0.0.1
```
Comment (jimmy0251) [inline go.mod:13]: This dependency is not used anymore. I think we should run `go mod tidy` to remove the dependencies from go mod.

File: contact/contact.go [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+	ContactType      string            `json:"contact_type"`
+}
+
+const CHARSET = "utf-8"
+
+type Template struct {
+	templates *template.Template
+}
+
+func New(TemplateFs embed.FS) *Template {
```
Comment (cp-sumi-k) [inline contact/contact.go:39]: Make variable name start with small letter

File: contact/contact.go [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+		log.Fatal(err)
+		return nil, err
+	}
+
+	return sess, err
+
+}
+
+func GetEmailTemplate(htmlBody string, data ContactDetails, title string, sender string, receiver string) (template *ses.SendEmailInput) {
+	contactData := ConcatenateContactData(data)
```
Comment (cp-sumi-k) [inline contact/contact.go:142]: We do not need to concat every details of contact.
Just add string like "Contact Info" as data

**Review by cp-sumi-k — APPROVED** [review]

**Pull request merged**